code/GatewayFieldsFactory.php

```diff
@@ -320,47 +340,118 @@ protected function cullForGateway(&$fields, $defaults = array())
      *
      * @return array
      */
-    public function normalizeFormData(array $data) {
-
-        $renameMap = \Config::inst()->get('GatewayFieldsFactory', 'rename');
-        
-        if (!$renameMap || empty($renameMap)) {
+    public function normalizeFormData(array $data)
+    {
+        if (empty($data)) {
             return $data;
         }
 
-        $hasPrefix = false;
+        $renameMap = array_flip(static::$renamemap);
 
-        if (array_key_exists('prefix', $renameMap)) {
-            $hasPrefix = true;
+        foreach ($renameMap as $customName => $defaultName) {
+            if (array_key_exists($customName, $data)) {
+                $value = $data[$customName];
+                unset($data[$customName]);
+                $data[$defaultName] = $value;
+            }
         }
 
-        // Fix for: array_flip(): Can only flip STRING and INTEGER values!
-        // This would occur if you have defined Gateways in rename.yml
-        foreach($globalMap = $renameMap as $key => $value) {
-            if (is_array($value)) {
-                unset($globalMap[$key]);
-            }
+        return $data;
+    }
+
+    /**
+     * Fetches the prefix (if defined) from the global section of the rename map
+     * @return mixed
+     */
+    public function getGlobalPrefix()
```

Comment:
If we use the approach to pre-calculate everything within `buildRenameMap`, this method can be removed. Or do you think it would be useful as a public API?